rust/connlib/shared/src/messages.rs

```diff
@@ -203,6 +203,18 @@ pub struct ResourceDescriptionCidr {
     pub name: String,
 }
 
+#[derive(Debug, Deserialize, Serialize, Clone, PartialEq, Eq)]
+#[serde(tag = "protocol", rename_all = "snake_case")]
+#[non_exhaustive]
```

Comment:
Is there much benefit to this? We have a mono-repo so we can update all the Rust code in one go.

`non_exhaustive` is typically used for API stability reasons to be able to make non-breaking changes. We don't release any of our stuff to crates.io so I don't really see the value in this.

In fact, it might even hurt a bit because it means we cannot rely on the Rust compiler to tell us all the places we need to touch when we add a variant.

Reply:
I was thinking that it it forces us to match for not yet existent variants, in case we want to add some in the future without breaking api compatibility with the portal.

I see now that this doesn't work without telling serde to properly deserialize, still when none of the variants match.

I'll just remove it for now.